python/lib/eeg.py

```diff
@@ -687,6 +705,99 @@ def fetch_and_insert_event_file(
 
         return event_path
 
+    def fetch_and_insert_annotation_files(
+            self, physiological_file_id, derivatives=None):
+        """
+        Gather raw channel file information to insert into
+        the physiological_annotation_* tables. Once all the information has been gathered,
+        it will call Physiological.insert_annotation_file that will perform the
+        insertion into the physiological_annotation_* tables, linking them to the
+        PhysiologicalFileID already registered.
+
+        :param physiological_file_id: PhysiologicalFileID of the associated
+                                      physiological file already inserted into
+                                      the physiological_file table
+         :type physiological_file_id: int
+        :param derivatives: dictionary with derivative folder information if
+                            the event file to insert is a derivative file.
+                            Set by default to None when inserting raw file.
+         :type derivatives: dict
+
+        :return: channel file path in the /DATA_DIR/bids_import directory
+         :rtype: str
+        """
+
+        # load the Physiological object that will be used to insert the
+        # physiological data into the database
+        physiological = Physiological(self.db, self.verbose)
+
+        # check if inserting derivatives to use the derivative_pattern to
+        # grep for the eeg file
+        derivative_pattern = None
+        derivative_path    = None
+        if derivatives:
+            derivative_pattern = derivatives['derivative_name'] + "/sub-"
+            derivative_path    = self.get_derivatives_path(derivatives)
+
+        annotation_metadata_file = BidsReader.grep_file(
+            files_list         = self.annotations_files,
+            match_pattern      = 'annotations.json',
+            derivative_pattern = derivative_pattern
+        )
+
+        annotation_data_file = BidsReader.grep_file(
+            files_list         = self.annotations_files,
+            match_pattern      = 'annotations.tsv',
+            derivative_pattern = derivative_pattern
+        )
+
+        if not (annotation_metadata_file) or not(annotation_data_file):
+            message = "WARNING: no annotations files associated with " \
+                      "physiological file ID " + str(physiological_file_id)
+            print(message)
+            return None
+        else:
+            annotation_paths = self.db.pselect(
```

Comment:
the select statement should be in a function in `lib/database_lib/physiologicalannotationfile.py` file which will return the annotation_path. To keep all the database calls into the `database_lib` folder.